// node-gpt-api/dist/api.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.deleteTask = exports.waitAndGetTask = exports.getTask = exports.createTask = exports.ask = void 0;
let api_origin = 'http://localhost:8041';
async function ask(question, callback) {
    let id = await createTask(question);
    if (!callback) {
        return waitAndGetTask(id);
    }
    for (;;) {
        let task = await getTask(id);
        await callback(task);
        if (task.completed) {
            return task;
        }
    }
}
exports.ask = ask;
async function createTask(question) {
    let res = await fetch(`${api_origin}/tasks`, {
        method: 'POST',
        headers: {
            'Content-Type': 'application/json',
            'Accept': 'application/json',
        },
        body: JSON.stringify({ question }),
    });
    let json = await res.json();
    if (json.error)
        throw new Error(json.error);
    return json.id;
}
exports.createTask = createTask;
async function getTask(id) {
    let res = await fetch(`${api_origin}/tasks/${id}`, {
        headers: { Accept: 'application/json' },
    });
    let json = await res.json();
    if (json.error)
        throw new Error(json.error);
    return json.task;
}
exports.getTask = getTask;
async function waitAndGetTask(id) {
    let res = await fetch(`${api_origin}/tasks/${id}?completed=true`, {
        headers: { Accept: 'application/json' },
    });
    let json = await res.json();
    if (json.error)
        throw new Error(json.error);
    return json.task;
}
exports.waitAndGetTask = waitAndGetTask;
async function deleteTask(id) {
    let res = await fetch(`${api_origin}/tasks/${id}`, {
        method: 'DELETE',
        headers: { Accept: 'application/json' },
    });
    let json = await res.json();
    if (json.error)
        throw new Error(json.error);
}
exports.deleteTask = deleteTask;
